pkg/apis/core/validation/validation.go

```diff
@@ -4921,6 +4927,12 @@ func ValidateSecurityContext(sc *core.SecurityContext, fldPath *field.Path) fiel
 		}
 	}
 
+	if sc.RunAsGroup != nil {
+		if validation.IsValidGroupID(*sc.RunAsGroup) != nil {
```

Comment:
same question here, if RunAsGroup is non-nil, do we require RunAsUser to be non-nil?

Reply:
same RunAsGroup can be non nil and RunAsUser can be nil

Reply:
Is that because the requirement for runasuser to be non-nil is docker specific? If other runtimes could allow that, it's fine to skip validating that here.

Reply:
the validation is saying that if RunAsGroup is specified at all, then it should be a valid group id. Not sure i follow your comment. Its not validating whether its specified or not. The first if is validating whether its specified at all, the second if is validating that if its specified its a valid group id

Reply:
I was wondering if we wanted to validate that both run as user and run as group were set, because the docker runtime will error if you set group without setting user

Reply:
yes other implementations could allow that and the limitation is docker specific.  for e.g. rkt allows setting both independently

Reply:
There's a validation for it in dockershim - it might be prudent to push that up to the API validation for now.